app/views/posts/_list_item.haml

```diff
@@ -11,9 +11,14 @@
     - if !current_user.try(:hide_warnings) && post.content_warnings.exists?
       = image_tag "/images/exclamation.png", class: 'vmid', title: "Content Warning: " + post.content_warnings.pluck(:name).join(', '), alt: '!'
   %td.post-subject{class: klass}
-    - if logged_in? && (@show_unread || (!@unread_ids.nil? && @unread_ids.include?(post.id)))
+    - if logged_in? && ((@opened_ids.nil? || !@opened_ids.include?(post.id)) || (!@unread_ids.nil? && @unread_ids.include?(post.id)))
```

Comment:
Oh, oops – I think this line should use `(!@opened_ids.nil? && !@opened_ids.include?(post.id))`